fivetran-trigger/main.py

```diff
@@ -19,48 +25,105 @@ def init():
+def setup_logging():
+    """
+    Sets up logging for the application.
+    """
+    handler = logging.StreamHandler(stream=sys.stdout)
+    formatter = CloudLoggingFormatter(fmt="[%(name)s] %(message)s")
+    handler.setFormatter(formatter)
+    logger = logging.getLogger("primary_logger")
+    logger.addHandler(handler)
+    logger.setLevel(logging.DEBUG)
+    sys.excepthook = handle_unhandled_exception
+
+
+def handle_unhandled_exception(exc_type, exc_value, exc_traceback):
+    """
+    Handles unhandled exceptions by logging the exception details and sending an alert to the development team.
+
+    This function is intended to be used as a custom excepthook function, which is called when an unhandled exception
+    occurs in the application. The function logs the exception details to the primary logger, and sends an alert to
+    the development team using a third-party service such as Datadog or PagerDuty.
+
     Args:
-        request (flask.Request): The request object.
-        <https://flask.palletsprojects.com/en/1.1.x/api/#incoming-request-data>
-    Returns:
-        The response text, or any set of values that can be turned into a
-        Response object using `make_response`
-        <https://flask.palletsprojects.com/en/1.1.x/api/#flask.make_response>.
+        exc_type (type): The type of the exception that was raised.
+        exc_value (Exception): The exception object that was raised.
+        exc_traceback (traceback): The traceback object that was generated when the exception was raised.
     """
+    if issubclass(exc_type, KeyboardInterrupt):
+        sys.__excepthook__(exc_type, exc_value, exc_traceback)
+        return
+    # Log the unhandled exception
+    logger = logging.getLogger("primary_logger")
+    logger.exception(
+        "Unhandled exception", exc_info=(exc_type, exc_value, exc_traceback)
+    )
+    # Send an alert to the development team using a third-party service such as Datadog or PagerDuty
+    # TODO: Add code to send an alert to the development team
+
+
+@functions_framework.http
+def trigger_sync(request):
+    """
+    Triggers a Fivetran sync for a given connector ID.
+
+    """
+    setup_logging()
+    init()
+
+    # Try get_json first
     request_json = request.get_json(silent=True)
-    request_args = request.args
 
-    if request_json and "name" in request_json:
-        name = request_json["name"]
-    elif request_args and "name" in request_args:
-        name = request_args["name"]
-    else:
-        name = "World"
-    message = f"Hello {escape(name)}!"
-    endpoint = "anything"
-    url = f"https://httpbin.org/{endpoint}"
-    payload = {"message": message}
-    resp = session.request(
-        "POST",
-        url,
-        headers={
-            "Content-Type": "application/json",
-            "Accept": "application/json;version=2",
-        },
-        json=payload,
-        auth=basic_auth,
-    )
+    # If that fails and we have octet-stream content type, try manual parsing
```

Comment:
I suspect we can fix the content-side with the scheduler job config. But let's get this in prod first. Good workaround.